conan/tools/microsoft/msbuilddeps.py

```diff
@@ -65,20 +64,40 @@ class MSBuildDeps(Generator):
         </Project>
         """)
 
-    @property
-    def filename(self):
-        return None
-
-    @ staticmethod
-    def _name_condition(settings):
-        props = [("Configuration", settings.build_type),
-                 # FIXME: This probably requires mapping ARM architectures
-                 ("Platform", {'x86': 'Win32',
-                               'x86_64': 'x64'}.get(settings.get_safe("arch")))]
-
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.configuration = conanfile.settings.build_type
+        self.platform = {'x86': 'Win32',
+                         'x86_64': 'x64'}.get(str(conanfile.settings.arch))
+        # TODO: this is ugly, improve this
+        self.output_path = os.getcwd()
+
+    def generate(self):
+        # TODO: Apply config from command line, something like
+        # configuration = self.conanfile.config.generators["msbuild"].configuration
+        # if configuration is not None:
+        #     self.configuration = configuration
+        # platform
+        # config_filename
+        # TODO: This is duplicated in write_generators() function, would need to be moved
+        # to generators and called from there
```

Comment:
We can check here that `self.configuration` and `self.platform` are different from `None`

Reply:
I was adding the check:
```python
        if self.configuration is None:
            raise ConanException("MSBuildDeps.configuration is None, it should have a value")
        if self.platform is None:
            raise ConanException("MSBuildDeps.platform is None, it should have a value")
```
Then realized that both ``condition`` and ``configfilename`` already provide for a ``None`` value, filtering it. This might make sense in single-config environments (we have been reported that MSBuild can be used for Linux builds), so maybe it is not worth raising an exception?

Reply:
IIRC, VS IDE uses something like `AnyCPU` or `AnyPlatform` if the same configuration should apply for `x86` and `x64` (I cannot remember what was it like for the Configuration)... so it is assigning a value to the variable that could be evaluated in the _Condition_. I cannot check because I don't have a running Visual Studio.

I don't know which one is the right approach but if the IDE always assigns a value to those properties (Configuration and Platform) maybe it is better to take those values into account instead of ignoring them in the Condition... if that is the scenario, the user would need to assign `msbuild.platform = 'AnyCPU'`. 

No idea how it looks like for those Linux build, never seen them. Maybe there is strong evidence and all we need is a test.

Reply:
You are right, it seems impossible to remove or leave empty this field, you can rename to a custom config, but you need to have one. Lets add the exceptionst then.